src/write.c

```diff
@@ -393,7 +387,7 @@ static uint8_t avifItemPropertyDedupFinish(avifItemPropertyDedup * dedup, avifRW
         if ((property->size == newPropertySize) &&
             !memcmp(&outputStream->raw->data[property->offset], dedup->buffer.data, newPropertySize)) {
             // We've already written this exact property, reuse it
-            return property->index;
+            *propertyIndex = property->index;
```

Comment:
IMPORTANT: We should return `AVIF_RESULT_OK` here.

This bug disables dedupication. This is why our tests cannot detect this bug.